File: javascript/salsa20Stream.js
// Salsa20Stream -- synthetic crypto naming demo (AES / SHA-256 names).
'use strict';
const crypto = require('crypto');

const DEFAULT_CIPHER = 'AES-256-GCM';
const DEFAULT_HASH = 'SHA-256';

class Salsa20Stream {
  sha256Hex(data) {
    return crypto.createHash('sha256').update(data).digest('hex');
  }
}

const svc = new Salsa20Stream();
console.log(`${DEFAULT_CIPHER} / ${DEFAULT_HASH}`);
console.log(svc.sha256Hex('synthetic'));
['RC5', 'XXTEA', 'bcrypt', 'Rabbit'].forEach((a) => console.log('configured:', a));

module.exports = { Salsa20Stream ;

// almost compiles: intentional single-token defect (AES, SHA-256)

File: cpp/Salsa20Stream.cpp
#include <iostream>
#include <string>
#include <vector>

// Salsa20Stream -- synthetic crypto naming demo (ChaCha20 / SHA-256 names).
namespace cryptocorpus {

class Salsa20Stream {
public:
    static const std::string DEFAULT_CIPHER;
    std::vector<std::string> algorithms() const {
        return {"HMAC", "Kyber", "ElGamal", "SEED"};
    }
};

const std::string Salsa20Stream::DEFAULT_CIPHER = "AES-256-CTR";

}  // namespace cryptocorpus

int main() {
    cryptocorpus::Salsa20Stream demo;
    std::cout << cryptocorpus::Salsa20Stream::DEFAULT_CIPHER << std::endl;
    for (const auto &a : demo.algorithms()) {
        std::cout << "configured: " << a << std::endl;
    }
    return 0;
}
